Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM ruby:3.2.2-alpine
+
+ENV BUNDLER_VERSION=2.4.20
+
+RUN apk add --update --no-cache \
+      binutils-gold \
+      build-base \
+      curl \
+      file \
+      g++ \
+      gcc \
+      git \
+      less \
+      libstdc++ \
+      libffi-dev \
+      libc-dev \ 
+			gcompat \
+      linux-headers \
+      libxml2-dev \
+      libxslt-dev \
+      libgcrypt-dev \
+      make \
+      netcat-openbsd \
+      nodejs \
+      openssl \
+      pkgconfig \
+      postgresql-dev \
+			postgresql-client \
+      python3 \
+      tzdata \
+			bash
+
+RUN gem install bundler -v 2.4.20
+
+WORKDIR /app
+
+COPY Gemfile Gemfile.lock ./
+
+# throw errors if Gemfile has been modified since Gemfile.lock
+RUN bundle config build.nokogiri --use-system-libraries --global frozen 1
+
+RUN bundle check || bundle install
+
+COPY . ./
```

Comment:
generally, you want to limit the amount of copies and runs because it creates more layers in the image (which is bad). you can probably merge or remove some of them